gh-127495: Append to history file after every statement in PyREPL (#132294)
python/cpython: Misc/NEWS.d/next/Library/2025-04-08-14-50-39.gh-issue-127495.Q0V0bS.rst

> Save (append new entry) the repl history to file every time new statement
> was succesful.  This should preserve command-line history after interpreter
> crash. Patch by Sergey B Kirpichev.

Reviewer: "Successful" is misleading here (and a typo). In your test `time.sleep()` doesn't successfully finish executing, but it's put in history already.

```suggestion
In PyREPL, append a new entry to the ``PYTHON_HISTORY`` file after every
statement. This should preserve command-line history after interpreter
is terminated. Patch by Sergey B Kirpichev.
```

Author: > a typo

That's true :(

> In your test time.sleep() doesn't successfully finish executing, but it's put in history already.

No!  It shouldn't and added test checks that.

> Save (append new entry) the repl history to file every time new statement
> was successful.  This should preserve command-line history after interpreter
> crash. Patch by Sergey B Kirpichev.

Author: ```suggestion
In PyREPL, append a new entry to the ``PYTHON_HISTORY`` file after every
successfully finished statement.  This should preserve command-line history
after interpreter is terminated.  Patch by Sergey B Kirpichev.
```
